tests/ui/rfcs/rfc-0000-never_patterns/parse.rs

```diff
@@ -0,0 +1,71 @@
+#![feature(never_patterns)]
+#![allow(incomplete_features)]
+
+enum Void {}
+
+fn main() {}
+
+macro_rules! never {
+    () => { ! }
+}
+
+fn parse(x: Void) {
+    match None::<Void> {
+        None => {}
```

Comment:
What happens when we do `! | _`?

e.g.

```
match None::<!> {
  Some(!) | _
```

Reply:
Actually, I misinterpreted what `is_never_pattern` was doing. Okay, that makes sense!